### PR TITLE
fix: Week 1 product polish — bugs, error handling, and docs

### LINKED ISSUE #1: Week 1: Polish & Prepare
## Product (3 hrs)

- [x] Fix any critical bugs in current codebase
- [x] Add basic error handling for edge cases
- [x] Test installation flow on fresh machine
- [ ] Document any gotchas in README

## Content (4 hrs)

- [x] Customize launch blog post draft
  - Add personal details to story
  - Replace all placeholders (GitHub URL, blog URL, Twitter)
  - Edit for your voice/tone
  - Proofread thoroughly
- [x] Create demo video script (outline what to show)
- [x] Take 5-10 screenshots of key features for README

## Setup (2 hrs)

- [x] Enable GitHub Sponsors
  - Create tiers: $5 Coffee, $25 Supporter, $50 Advocate
  - Write tier descriptions focusing on community support
  - Add sponsor button to repository
- [ ] Set up Twitter/X if not already active
- [x] Prepare launch post for Reddit (draft in docs)

**Goal**: Everything ready to launch

### PR DESCRIPTION
**Pull request by jamesfishwick**
## Summary

Addresses the Product checklist from issue #1 (Week 1: Polish & Prepare).

- **fix: `~` not expanded in config paths** — Claude Desktop passes env vars literally from JSON; `~` never shell-expands. `config.py` now calls `os.path.expanduser()` on all three path fields so users who copy the README example verbatim get the correct home-relative directory.
- **fix: FTS5 `OperationalError` crashes search** — `search_by_text()` and `search_combined()` had no error handling around `session.execute()`. Malformed FTS5 syntax (even after escaping) raised a raw SQLAlchemy traceback. Now caught and logged; returns `[]` gracefully.
- **fix: parameter validation in 3 MCP tools** — `zk_find_similar_notes`, `zk_find_central_notes`, and `zk_get_cluster_report` now validate `threshold`/`min_score` ∈ [0.0, 1.0] and `limit > 0`, returning a clear error string for invalid input.
- **docs: README install instructions** — Fixed placeholder clone URL, replaced `~` with full absolute paths in JSON config example, added `pipx environment --value VENV` command to locate the Python binary, added four new Troubleshooting entries.

## Test plan

- [ ] All 92 tests pass: `uv run pytest tests/ --tb=short -q`
- [ ] Set `ZETTELKASTEN_NOTES_DIR=~/.local/share/...` — verify server creates the correct expanded path, not a literal `~/...` directory
- [ ] Search with an unclosed FTS5 quote — verify empty results, no exception
- [ ] Call `zk_find_similar_notes` with `threshold=2.0` — verify error string returned
- [ ] Call `zk_find_central_notes` with `limit=-1` — verify error string returned

Closes #1 (Product section)

### CODE REVIEW

File: src/zettelkasten_mcp/server/mcp_server.py [hunk 1]
```diff
@@ -472,6 +472,10 @@ def zk_find_similar_notes(
                 limit: Maximum results (default: 5)
             """
             try:
```
Comment (Korgull-the-Censorious) [inline src/zettelkasten_mcp/server/mcp_server.py]: **RECOMMENDED: log validation rejections for observability**

The guard clause returns an error string when `threshold` is out of range, but nothing is logged. If an LLM client or integration layer consistently passes invalid parameters, there is no signal in the logs.

```python
if not 0.0 <= threshold <= 1.0:
    logger.warning("zk_find_similar_notes: threshold %r out of range [0.0, 1.0]", threshold)
    return "Error: threshold must be between 0.0 and 1.0."
```

Same applies to the `limit` check here and to the guards in `zk_find_central_notes` and `zk_get_cluster_report`.

**Severity:** RECOMMENDED — no behavior change, adds diagnostic visibility

File: src/zettelkasten_mcp/config.py [hunk 2]
```diff
@@ -9,16 +9,16 @@
 class ZettelkastenConfig(BaseModel):
     """Configuration for the Zettelkasten server."""
     base_dir: Path = Field(
-        default_factory=lambda: Path(os.getenv("ZETTELKASTEN_BASE_DIR", "."))
+        default_factory=lambda: Path(os.path.expanduser(os.getenv("ZETTELKASTEN_BASE_DIR", ".")))
     )
```
Comment (Korgull-the-Censorious) [inline src/zettelkasten_mcp/config.py]: **RECOMMENDED: add a post-expansion check for unexpanded `~username` patterns**

`os.path.expanduser('~nonexistentuser')` returns the input unchanged rather than raising. If a user sets `ZETTELKASTEN_NOTES_DIR=~baduser/notes`, the server silently writes notes to `./~baduser/notes/` relative to CWD — wrong location, no error.

A bare `~` always expands correctly; the risk is `~othername/...` patterns. A startup assertion catches this before notes are lost:

```python
notes_dir: Path = Field(
    default_factory=lambda: _resolve_path(os.getenv("ZETTELKASTEN_NOTES_DIR", "data/notes"))
)

def _resolve_path(raw: str) -> Path:
    expanded = os.path.expanduser(raw)
    if expanded.startswith("~"):
        raise ValueError(f"Path {raw!r} could not be expanded — home directory not found. Use an absolute path.")
    return Path(expanded)
```

**Severity:** RECOMMENDED — narrow edge case, but the README change in this same PR increases the likelihood that users attempt `~`-prefixed paths

File: src/zettelkasten_mcp/services/search_service.py [hunk 3]
```diff
@@ -265,7 +269,11 @@ def search_combined(
                 WHERE notes_fts MATCH :query
                 ORDER BY bm25(notes_fts)
             """)
```
[the remark this answers is not in the document]
Comment (Korgull-the-Censorious) [reply src/zettelkasten_mcp/services/search_service.py]: Agreed on both counts. Fixed: same `OperationalError`/`"fts5"` narrowing, and on FTS5 syntax error the method now falls back to returning the already-computed metadata-filtered candidates as unranked `SearchResult`s (score=0.0, empty context) rather than discarding them. Non-FTS5 `OperationalError`s re-raise.

**Review by Korgull-the-Censorious — CHANGES_REQUESTED** [review]
The ruins have shifted. A rename merged while this branch slept.

BLOCKING:
- src/zettelkasten_mcp/ → src/slipbox_mcp/ on main (PR #9). This entire diff targets dead code paths. Rebase required.

REQUIRED:
- search_service.py: FTS5 error string check misses 'unterminated string' and 'no such column' — expand the pattern
- search_service.py search_combined: fallback returns results indistinguishable from a successful search — add degraded-mode context string

These findings are annotated inline.

What survives the rubble: _expand_path() logic is correct, OperationalError narrowing is architecturally sound, logger.warning placement is correct. The intent is good. The execution needs a rebase and two targeted fixes.

Rebuild on current terrain. The Exterminator awaits.

File: src/slipbox_mcp/services/search_service.py [hunk 4]
```diff
@@ -265,7 +273,26 @@ def search_combined(
...
+                    logger.warning(
+                        "FTS5 query syntax error for %r, returning metadata-only results: %s",
+                        fts_query, e,
+                    )
+                    notes = [repository._db_note_to_note(n) for n in db_notes]
+                    return [
+                        SearchResult(
+                            note=n,
+                            score=0.0,
+                            matched_terms=set(query_text.split()) if query_text else set(),
```
Comment (Korgull-the-Censorious) [inline src/slipbox_mcp/services/search_service.py]: **IMPORTANT: `matched_terms` in fallback claims terms were matched — they were not**

Setting `matched_terms=set(query_text.split())` in the FTS5 fallback path is misleading. These terms were never matched — the text search failed. Any caller using `matched_terms` for result highlighting will incorrectly highlight words that did not appear in search results.

The `if query_text else set()` guard is also dead code: `search_combined` returns early before this path is reachable when `query_text` is falsy.

**Recommendation:**

```python
matched_terms=set(),
```

`matched_context` already signals degraded mode. `matched_terms` should be empty.

**Severity:** IMPORTANT — misleads callers using matched_terms for highlighting

**Review by Korgull-the-Censorious — APPROVED** [review]
The nuclear dust settles. The exterminator finds... acceptable work.

✓ Tilde expansion handled at both config layer (_expand_path) and README documentation
✓ FTS5 OperationalError properly triaged: syntax errors return gracefully, schema corruption re-raises
✓ Missing-table log messages now distinguish 'notes_fts' from other tables
✓ ValidationError caught alongside ValueError at config startup — no raw tracebacks
✓ Parameter guard clauses on three MCP tools (threshold, limit, min_score) with correct boundaries
✓ 19 tests added covering all new code paths; vacuous assertions eliminated
✓ 111 tests passing

The code may endure. APPROVED.

File: src/slipbox_mcp/services/search_service.py [hunk 5]
```diff
@@ -265,7 +272,20 @@ def search_combined(
                 WHERE notes_fts MATCH :query
                 ORDER BY bm25(notes_fts)
             """)
```
[the remark this answers is not in the document]
Comment (james-livefront) [reply src/slipbox_mcp/services/search_service.py]: Fixed. The fallback now sets matched_terms=set() and matched_context to a descriptive string ('(text search unavailable — showing metadata matches only)') so callers can distinguish degraded mode from a successful empty search. The metadata-filtered candidates are preserved and returned as SearchResult objects with score=0.0.
